### ISSUE TITLE
fastcat::Manager::GetDeviceStatePointers dynamically allocates memory

### ISSUE THREAD

**Issue by d-loret** [nasa-jpl/fastcat#30]
Unlike `fastcat::Manager::GetDeviceStates`, `fastcat::Manager::GetDeviceStatePointers` dynamically allocates memory each time it is called. The vector is constructed and resized with each call of the function. This makes the function non real-time safe.

**Comment by alex-brinkman** [comment]
The idea behind this function is to get fixed, shared pointers to device state data and that it would be called once by the application. Once the application has this array of pointer, it doesn't need to call this again and can just dereference it as needed to get the current state data.

I agree allocating the vector of array points is dynamically allocating memory which should be corrected. I think the best thing would be to pass a reference to this array like so:

``` c++
void fastcat::Manager::GetDeviceStatePointers(std::vector<std::shared_ptr<const fastcat::DeviceState>>& device_state_ptrs);
```
I'd probably overload this function name and asset if an application uses the version that dynamically allocates memory.

Do you agree with this fix?

**Comment by d-loret** [comment]
> pass a reference to this array 

Yes, that would be an easy way to fix it. It should probably be coupled with a function that allows the client application how many states are present in the manager so that it can pre-allocate properly once.
